zerver/models.py

```diff
@@ -748,20 +751,20 @@ def linebreak(string):
     # type: (str) -> str
     return string.replace('\n\n', '<p/>').replace('\n', '<br/>')
 
-def extract_message_dict(message_str):
-    # type: (str) -> Dict[str, Any]
-    return ujson.loads(zlib.decompress(message_str).decode("utf-8"))
+def extract_message_dict(message_bytes):
+    # type: (binary_type) -> Dict[text_type, Any]
+    return ujson.loads(zlib.decompress(message_bytes).decode("utf-8"))
 
 def stringify_message_dict(message_dict):
-    # type: (Dict[Any, Any]) -> str
-    return zlib.compress(ujson.dumps(message_dict).encode("utf-8"))
+    # type: (Dict[Any, Any]) -> binary_type
+    return zlib.compress(force_bytes(ujson.dumps(message_dict)))
```

Comment:
`stringify_message_dict` and `extract_message_dict` should be inverses of each other; maybe that should be reflected in the type of the input to stringify_message_dict?